int32 nGhosts
pos[] ghostsPos
int32[] mode
